react-native/src/App.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import {SafeAreaProvider} from 'react-native-safe-area-context';
+import {StyleSheet, Text, View} from 'react-native';
+import {NavigationContainer} from '@react-navigation/native';
+import {createStackNavigator} from '@react-navigation/stack';
+
+import {LogoutButton} from './LogoutButton';
+import {WelcomeView} from './WelcomeView';
+import {TasksView} from './TasksView';
+
+const Stack = createStackNavigator();
+
+const App = () => {
+  return (
+    <SafeAreaProvider>
+      <NavigationContainer>
+        <Stack.Navigator>
+          <Stack.Screen name="Welcome" component={WelcomeView} />
+          <Stack.Screen
+            name="Tasks"
+            component={TasksView}
+            options={{
+              headerLeft: () => {
+                return <LogoutButton />;
+              },
+            }}
+          />
+        </Stack.Navigator>
```

Comment:
I recommend adding some logic here to see if the current user has been set and basing the `Navigator` contents off that.
For best practice, I would follow the recommendations in the documentation for [Authentication flows](https://reactnavigation.org/docs/auth-flow/)
This would help prevent the UI from going back to the `Welcome` screen without an explicit logout.

Reply:
Decided against this for this app, considering we only have two screens. A big part of the reason is to keep the app simple, by avoiding useReducer or some other state management library, without which we'd have to explicitly ignore warnings such as "Non-serializable values were found in the navigation state".